### ISSUE TITLE
Gufuncs with zero inputs(!) only work with explicit output

### ISSUE THREAD

**Issue by mhvk** [numpy/numpy#15140]
I wouldn't quite dare to call this a bug, but it is a small inconsistency that perhaps leads to some refactoring that makes everything better. In the process of automatic wrapping of Standards of Astronomy functions (see #15139), some functions were wrapped that are meant to just give access to some standard numbers/positions/matrices. They are wrapped with signatures like `->(3,3)` and they work, but only if one gives an explicit output. If nothing is given, the iterator raises `ValueError: At least one iterator operand must be non-NULL`. 

This of course is not illogical, but in principle, what should happen here is quite obvious, so perhaps it can nevertheless be remedied...

**Comment by eric-wieser** [comment]
I assume you're suggesting broadcast dimensions should be inferred as `()`

**Comment by mhvk** [comment]
Yes, indeed. A bit like assuming there always is an otherwise irrelevant scalar input present (the "identity" of broadcasting?)

**Comment by eric-wieser** [comment]
I think `np.broadcast` already does that
